lib/spack/spack/compilers/__init__.py

```diff
@@ -125,6 +125,105 @@ def get_compiler_config(scope=None, init_config=True):
     return config
 
 
+def get_compiler_config_from_packages(scope=None):
+    """Return the compiler configuration from packages.yaml"""
+    packages = []
+    config = spack.config.get("packages", scope=scope)
+
+    if not config:
+        return packages
+
+    for name, entry in config.items():
+        if name not in supported_compilers() and name not in package_name_to_compiler_name.keys():
```

Comment:
(minor) We can compute the possible compiler packages once outside of the loop. That would be slightly more readable:
```suggestion
    possible_compiler_packages = supported_compilers() + list(package_name_to_compiler_name)
    for name, entry in config.items():
        if name not in possible_compiler_packages:
```